File: python/binary_search.py
def binary_search(element_list, target, start, end):
    if start > end:
        return
    mid_point = (start + end) // 2
    if element_list[mid_point] > target:
        return binary_search(element_list, target, start, mid_point-1)
    elif element_list[mid_point] < target:
        return binary_search(element_list, target, mid_point+1, end)
    else:
        return mid_point

File: ruby/Mixed/binary_search.rb
def b_search(arr, target, min = 0, max = arr.size - 1)
  return 'NONE' if min > max

  mid_point = ((min + max) / 2).to_i
  if arr[mid_point] > target
    return b_search(arr, target, min, mid_point - 1)
  elsif arr[mid_point] < target
    return b_search(arr, target, mid_point + 1, max)
  else
    return mid_point
  end
end
